src/app/InteractionModelDelegate.h

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the delegate zcl/app needs to implement, which can talk to interaction model.
+ *
+ */
+
+#pragma once
+
+#include <core/CHIPCore.h>
+#include <messaging/ExchangeContext.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace app {
+
+struct EventPathParams;
+
+/**
+ * @brief
+ *   This class provides a delegate implemented by zcl/app-level code
+ */
+class InteractionModelDelegate
+{
+public:
+    /**
+     * The zcl/app-level is expected to fill in the event paths that it wants to
+     * read to.
+     *
+     * @param[out]  aEventPathParamsList       a list of event paths the read client is interested in, zcl/app can
+     *                                         allocate path list in heap, and zcp/app would free the allocated path after read
+     *                                         interaction complete
```

Comment:
it could be stack or heap allocated, SDK consumer is responsible for its lifecycle. 
if I send a read request I have to allocate memory for the paths, after SendReadRequest is invoked, SDK consumer can decide if it can be deallocated. If readRequest fails to send out or response timeout, SDK consumer may decide to resend read request with previous paths.

Reply:
After more consideration, I decide to remove this callback and put event list as one parameter for send read request function.
I envision that
subscribe request paths may be initialized in the beginning and would not be changed over time, and we need this callback to know the interested attribute/event path, particularly when it resubscribe, internally it can fetch path via this callback.
In contrast, for read request, we don't have the requirement as the above re-subscribe has, so we just need send read request along with path directly.